# Taller de visualización de datos, "Visualizando datos de COVID-19
## Manejo y procesamiento de datos
En esta sección del tutorial descragaremos los datos, los procesaremos, haremos algunas gráficas exploratorias y generaremos los insumos necesarios para visualizar nuestros datos en la web.

La celda que estas leyendo es una celda de texto, no es necesario que teclees todo esto en tu ejemplo. La siguiente celda es una celda de código. Para seguir el ejemplo tendras que ir tecleando cada una de ellas y ejecutandolas paso a paso. La manera más fácil de ejecutar una celda de Jupyter es después de teclearla presionar Shift+Enter, esto ejecutará el código en ella, te mosdtrará el resultado de la ejecución si es que hay algo que mostrar y te dará una celda nueva debajo de la celda actual. Probemos con la siguiente celda, escribe 2+2 y presiona Shift+Enter, la celda debería de darte el resultado de la suma y colocarte en una celda nueva.

In [3]:
2+2

4

Si se ejecutó la celda y obtuviste el resultado todo está listo para continuar.

## Importando las librerías necesarias para realizar análisis de datos

Lo primero que haremos es importar las librerías necesarias para hacer análisis de datos. Por lo pronto necesitaremos 2 librterías:

* [Pandas](https://pandas.pydata.org/pandas-docs/stable/index.html)
* [Bokeh](https://docs.bokeh.org/en/latest/index.html)

Pandas lo usaremos para manipular de formas muy eficientes nuestra base de datos. 

Bokeh lo utilizaremos para graficar y realizar algunas visualizaciones de datos básicas con nuestros datos.

Ambas las importaremos a nuestro notebook a traves del comando import y las guardaremos en aliases para que no tengamos que teclear tanto cuando las llamamos. Pandas la guardaremos en el alias "pd" y Bokeh lo guardaremos en el alias "bpl"


In [4]:
import pandas as pd 
import bokeh.plotting as bpl

## Descargando los datos

En este tutorial trabajaremos con los datos mexicanos reportados por la secretaría de salud relacionados al COVID19. Los datos se publican diariamente y se pueden encontrar [aquí](https://www.gob.mx/salud/documentos/datos-abiertos-152127).

En específico nos concentraremos en los datos actuales, que se encuentran en un enlace en esa página. 

Para poder trabajar con los datos en python necesitamos "cargarlos" a memoria. Pandas nos permite crear una estructura de datos con ellos. A esta estructura la llamamos DataFrame.

Pandas nos permite crear DataFrames a partir de una gran cantidad de formatos de datos. Los más usuales son los formatos de texto separado por comas o "CSVs", texto en notación de objeto tipo JavaScript o "JSON", hojas de cálculo de excel. Además Pandas nos permite indicar la dirección de donde está guardado nuestro archivo a utilizar, puede ser una dirección local en tu computadora, por ejemplo una carpeta donde tengas guardados los datos o una dirección URL donde estén almacenados los datos en la web.

Para este ejercicio descargaremos directamente los datos de la página de la secretaría de salud pública. Los datos que liberaron están en formato zip, dentro del archivo comprimido hay un archivo de tipo CSV que podemos usar para crear un DataFrame. Pandas nos permite crear un DataFrame en nuestro notebook directamente desde la URL donde están guardados los datos, sin descargarlos y descomprimirlos a nuestra computadora previamente.

Teclea y ejecuta lo siguiente en tu notebook:

In [5]:
data = pd.read_csv('http://datosabiertos.salud.gob.mx/gobmx/salud/datos_abiertos/datos_abiertos_covid19.zip', compression='zip',encoding="latin1")

Hay varias cosas importantes en la celda anterior:

* Estamos ocupando la función read_csv de Pandas, por eso la invocamos como pd.read_csv() 
* El primer parámetro de la función es la ruta en la cual se encuentra nuestro archivo, en este caso es una dirección URL al archivo zip en la página de la secretaría de salud mexicana.
* Estamos utilizando la opción "compression" para indicar que es un archivo tipo zip
* Necesitamos indicar que la codificación del archivo es "latin1".
* La función read_csv() crea un DataFrame listo para procesarse. Este DataFrame lo guardamos en la variable "data".

Para poder visualizar nuestro DataFrame necesitamos ver el contenido de la variable data, nuestro DataFrame tiene la estructura de una tabla de datos, muy parecida a una hoja de cálculo. En la siguiente celda veremos que hay en esta variable

In [7]:
data

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
0                2020-10-18      1426fb       1       4          10     1   
1                2020-10-18      1c4583       2      12           9     2   
2                2020-10-18      0d55c9       2      12           9     1   
3                2020-10-18      071735       2       9          21     2   
4                2020-10-18      1468a5       1       4           5     1   
...                     ...         ...     ...     ...         ...   ...   
2183267          2020-10-18      332cf5       2       4          30     1   
2183268          2020-10-18      42c51a       2      12           9     2   
2183269          2020-10-18      408ef1       2       6           9     2   
2183270          2020-10-18      372368       2       9           8     2   
2183271          2020-10-18      427b3f       2       9          19     2   

         ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
0                 28           10              7              1  ...   
1                  9            9              4              1  ...   
2                  9            9             16              1  ...   
3                 21           21            114              1  ...   
4                 15            5             18              1  ...   
...              ...          ...            ...            ...  ...   
2183267           30           30            135              1  ...   
2183268            9            9             10              1  ...   
2183269            9            9              5              1  ...   
2183270            8            8             19              2  ...   
2183271           19           19             18              2  ...   

        RENAL_CRONICA TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  \
0                   2          2        99             1              1   
1                   2          2         2             1              1   
2                   2          2         2             1              1   
3                   2          2         2             1              1   
4                   2          2         1             1              1   
...               ...        ...       ...           ...            ...   
2183267             2          2         2             1              3   
2183268             2          2         1             1              3   
2183269             2          2         1             2             97   
2183270             2          2         2             1              3   
2183271             2          2         2             1              3   

         CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
0                          3        99            México           97   97  
1                          3        99            México           97   97  
2                          3        99            México           97   97  
3                          3        99            México           97   97  
4                          3        99            México           97   97  
...                      ...       ...                ...          ...  ...  
2183267                    6        99            México           97   97  
2183268                    6        99            México           97   97  
2183269                    1        99            México           97   97  
2183270                    6        99            México           97    2  
2183271                    6        99            México           97    2  

[2183272 rows x 38 columns]

Podemos observar que la estructura es muy similar a una hoja de cálculo. Cada dato está identificado por una columna en la parte posterior y un índice en la parte izquierda. Cada fila representa un caso en la base de datos, cada columna una variable con datos para cada caso. En la parte inferior podemos observar que nuestro DataFrame tiene 2,183,272 filas (este número puede variar dependiendo de cuando descargas el archivo pues diariamente este archivo crece debido a los casos que se agregan) y 38 columnas.

## Explorando un DataFrame

El primer paso para explorar una base de datos es conocer las variables que se encuentran en ella. En este caso podemos pedirle al DataFrame que nos muestre las columnas que contiene. El nombre de las columnas están guardadas en la propiedad "columns"

In [36]:
data.columns

Index(['FECHA_ACTUALIZACION', 'ID_REGISTRO', 'ORIGEN', 'SECTOR', 'ENTIDAD_UM',
       'SEXO', 'ENTIDAD_NAC', 'ENTIDAD_RES', 'MUNICIPIO_RES', 'TIPO_PACIENTE',
       'FECHA_INGRESO', 'FECHA_SINTOMAS', 'FECHA_DEF', 'INTUBADO', 'NEUMONIA',
       'EDAD', 'NACIONALIDAD', 'EMBARAZO', 'HABLA_LENGUA_INDIG', 'INDIGENA',
       'DIABETES', 'EPOC', 'ASMA', 'INMUSUPR', 'HIPERTENSION', 'OTRA_COM',
       'CARDIOVASCULAR', 'OBESIDAD', 'RENAL_CRONICA', 'TABAQUISMO',
       'OTRO_CASO', 'TOMA_MUESTRA', 'RESULTADO_LAB', 'CLASIFICACION_FINAL',
       'MIGRANTE', 'PAIS_NACIONALIDAD', 'PAIS_ORIGEN', 'UCI'],
      dtype='object')

Existen varias maneras para poder extraer pedazos o rebanadas del DataFrame. Por ejemplo podemos pedir una rebanada que contenga solamente la columna "FECHA_ACTUALIZACION". Para esto utilizaremos la notación de corchete, escribimos el nombre de la variable donde tenemos guardado nuestro DataFrame y luego entre corchetes la columna que nos interesa. La estructura resultante no es un DataFrame, es una Serie (similar a un DataFrame de una columna).

In [44]:
data["FECHA_ACTUALIZACION"]

0          2020-10-18
1          2020-10-18
2          2020-10-18
3          2020-10-18
4          2020-10-18
              ...    
2183267    2020-10-18
2183268    2020-10-18
2183269    2020-10-18
2183270    2020-10-18
2183271    2020-10-18
Name: FECHA_ACTUALIZACION, Length: 2183272, dtype: object

Tambien podemos pedir un conjunto de columnas, para ello utilizamos la misma notación pero con una lista de las columnas que deseamos, por ejemplo pidamos las columnas relacionadas con fechas:

In [49]:
data[["FECHA_ACTUALIZACION","FECHA_INGRESO","FECHA_SINTOMAS","FECHA_DEF"]]

FECHA_ACTUALIZACION FECHA_INGRESO FECHA_SINTOMAS   FECHA_DEF
0                2020-10-18    2020-03-31     2020-03-29  9999-99-99
1                2020-10-18    2020-03-30     2020-03-30  9999-99-99
2                2020-10-18    2020-03-26     2020-03-24  9999-99-99
3                2020-10-18    2020-03-18     2020-03-12  9999-99-99
4                2020-10-18    2020-04-02     2020-03-27  9999-99-99
...                     ...           ...            ...         ...
2183267          2020-10-18    2020-10-14     2020-10-11  9999-99-99
2183268          2020-10-18    2020-10-09     2020-10-07  9999-99-99
2183269          2020-10-18    2020-10-09     2020-10-02  9999-99-99
2183270          2020-10-18    2020-10-15     2020-10-05  9999-99-99
2183271          2020-10-18    2020-10-15     2020-10-10  9999-99-99

[2183272 rows x 4 columns]

La estructura de datos obtenida es un DataFrame con todas las filas de nuestro DataFrame original pero con solo 4 columnas.

También podemos pedir filas en específico, para distinguir entre columnas y filas siempre que necesitemos filas utilizaremos la propiedad "loc". Su uso es muy similar, por ejemplo, si quisera la fila 10 únicamente puedo hacerlo de la siguiente manera:

In [50]:
data.loc[10]

FECHA_ACTUALIZACION    2020-10-18
ID_REGISTRO                092521
ORIGEN                          1
SECTOR                         12
ENTIDAD_UM                      9
SEXO                            1
ENTIDAD_NAC                     9
ENTIDAD_RES                     9
MUNICIPIO_RES                  17
TIPO_PACIENTE                   1
FECHA_INGRESO          2020-03-30
FECHA_SINTOMAS         2020-03-28
FECHA_DEF              9999-99-99
INTUBADO                       97
NEUMONIA                        2
EDAD                           49
NACIONALIDAD                    1
EMBARAZO                        2
HABLA_LENGUA_INDIG              2
INDIGENA                        2
DIABETES                        2
EPOC                            2
ASMA                            2
INMUSUPR                        2
HIPERTENSION                    2
OTRA_COM                        2
CARDIOVASCULAR                  2
OBESIDAD                        2
RENAL_CRONICA                   2
TABAQUISMO    

También podemos pedir varias filas, por ejemplo hagamos un DataFrame con las filas 1,3 y 6:

In [54]:
data.loc[[1,3,6]]

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
1          2020-10-18      1c4583       2      12           9     2   
3          2020-10-18      071735       2       9          21     2   
6          2020-10-18      0e07d8       1       4          15     2   

   ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ... RENAL_CRONICA  \
1            9            9              4              1  ...             2   
3           21           21            114              1  ...             2   
6           15           15            104              2  ...             2   

  TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  CLASIFICACION_FINAL  \
1          2         2             1              1                    3   
3          2         2             1              1                    3   
6          2        99             1              1                    3   

   MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
1        99            México           97   97  
3        99            México           97   97  
6        99            México           97    2  

[3 rows x 38 columns]

La notación loc nos permite combinar y pedir solamente un número determinado de filas y un número determinado de columnas. Solamente tenemos que recordar que primero van las filas y luego las columnas, por ejemplo si quisieramos un mini DataFrame conformado por las filas 1,3,6 y las columnas relacionadas con fechas haríamos lo siguiente:

In [57]:
data.loc[[1,3,6],["FECHA_ACTUALIZACION","FECHA_INGRESO","FECHA_SINTOMAS","FECHA_DEF"]]

FECHA_ACTUALIZACION FECHA_INGRESO FECHA_SINTOMAS   FECHA_DEF
1          2020-10-18    2020-03-30     2020-03-30  9999-99-99
3          2020-10-18    2020-03-18     2020-03-12  9999-99-99
6          2020-10-18    2020-03-28     2020-03-28  2020-04-02

En el DataFrame hay variables muy interesantes, hay mucho material para hacer visualizaciones! Por el momento nos enfocaremos en los datos básicos reportados en las conferencias diarias de la secretaría de salud. La secretaría de Salud en su conferencia nocturna muestra la siguiente imagen:

![Resultados covid](./imagenes/img2.png)

En ella se muestran algunos datos que son importantes, principalmente el número de casos confirmados, el número de casos sospechosos, el número de casos negativos y el número de defunciones hasta la fecha. Los sospechosos los dividen en sospechosos sin muestra, sospechosos sin posibilidad de resultado y sopechosos con posibilidad de resultado.

Si analilzamos el diccionario de datos proporcionado por la Secretaría de Salud que se encuentra en este [enlace](http://datosabiertos.salud.gob.mx/gobmx/salud/datos_abiertos/diccionario_datos_covid19.zip) podemos observar que la variable "CLASIFICACION_FINAL" nos permite clasificar los casos de acuerdo a su calidad de confirmados, sospechosos o negativos. Para explorar los valores de una columna podemos utilizar la función value_counts()

In [11]:
data.groupby("CLASIFICACION_FINAL").size()

CLASIFICACION_FINAL
1      24032
2        619
3     826576
4       5778
5      40954
6     265492
7    1019821
dtype: int64

In [20]:
negativos = data[data["CLASIFICACION_FINAL"]==7]
negativos

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
3950             2020-10-18      13f641       1      12          26     1   
3951             2020-10-18      1e20d5       1       4          16     1   
3952             2020-10-18      01ee3f       1      12          24     2   
3953             2020-10-18      0cf214       1      12          13     2   
3954             2020-10-18      0b5ebb       2      12          19     1   
...                     ...         ...     ...     ...         ...   ...   
2134599          2020-10-18      38af8c       2       4           9     1   
2134600          2020-10-18      390ba7       1       4          14     1   
2134601          2020-10-18      33b9ba       2       4          21     1   
2134602          2020-10-18      33712d       2      12           9     2   
2134603          2020-10-18      374518       2      12           9     1   

         ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
3950              16           26             55              2  ...   
3951              16           16             52              1  ...   
3952              24           24             13              1  ...   
3953              13           13             60              1  ...   
3954              19           19             26              1  ...   
...              ...          ...            ...            ...  ...   
2134599            9            9             17              1  ...   
2134600           14           14             39              1  ...   
2134601           21           21            114              2  ...   
2134602           15            9              7              1  ...   
2134603            9            9             13              1  ...   

        RENAL_CRONICA TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  \
3950                2          2         2             1              2   
3951                2          2        99             1              2   
3952                2          2         2             1              2   
3953                2          2         2             1              2   
3954                2          2         2             1              2   
...               ...        ...       ...           ...            ...   
2134599             2          1         2             1              2   
2134600             2          2         1             1              2   
2134601             2          2         2             1              2   
2134602             2          1         1             1              2   
2134603             2          2         1             1              2   

         CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
3950                       7        99            México           97   99  
3951                       7        99            México           97   97  
3952                       7        99            México           97   97  
3953                       7        99            México           97   97  
3954                       7        99            México           97   97  
...                      ...       ...                ...          ...  ...  
2134599                    7        99            México           97   97  
2134600                    7        99            México           97   97  
2134601                    7        99            México           97    2  
2134602                    7        99            México           97   97  
2134603                    7        99            México           97   97  

[1019821 rows x 38 columns]

In [23]:
confirmados = data[data["CLASIFICACION_FINAL"].isin([1,2,3])]
confirmados

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
0                2020-10-18      1426fb       1       4          10     1   
1                2020-10-18      1c4583       2      12           9     2   
2                2020-10-18      0d55c9       2      12           9     1   
3                2020-10-18      071735       2       9          21     2   
4                2020-10-18      1468a5       1       4           5     1   
...                     ...         ...     ...     ...         ...   ...   
2183114          2020-10-18      2fcdc7       1      12          26     1   
2183136          2020-10-18      3c6f4d       1      12          26     2   
2183216          2020-10-18      373591       1       6          20     1   
2183262          2020-10-18      365e11       1       6           9     1   
2183269          2020-10-18      408ef1       2       6           9     2   

         ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
0                 28           10              7              1  ...   
1                  9            9              4              1  ...   
2                  9            9             16              1  ...   
3                 21           21            114              1  ...   
4                 15            5             18              1  ...   
...              ...          ...            ...            ...  ...   
2183114           26           26             30              1  ...   
2183136           26           26             19              1  ...   
2183216           20           20            399              1  ...   
2183262           17           15             58              2  ...   
2183269            9            9              5              1  ...   

        RENAL_CRONICA TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  \
0                   2          2        99             1              1   
1                   2          2         2             1              1   
2                   2          2         2             1              1   
3                   2          2         2             1              1   
4                   2          2         1             1              1   
...               ...        ...       ...           ...            ...   
2183114             2          2         1             2             97   
2183136             2          2         1             2             97   
2183216             2          2         2             2             97   
2183262             2          2         2             2             97   
2183269             2          2         1             2             97   

         CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
0                          3        99            México           97   97  
1                          3        99            México           97   97  
2                          3        99            México           97   97  
3                          3        99            México           97   97  
4                          3        99            México           97   97  
...                      ...       ...                ...          ...  ...  
2183114                    1        99            México           97   97  
2183136                    1        99            México           97   97  
2183216                    1        99            México           97   97  
2183262                    1        99            México           97    2  
2183269                    1        99            México           97   97  

[851227 rows x 38 columns]

In [29]:
sospechosos=data[data["CLASIFICACION_FINAL"]==6]
sospechosos

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
39125            2020-10-18      193243       2       4          19     2   
39126            2020-10-18      0462c9       2       4          14     1   
39127            2020-10-18      0cafed       2       4          19     2   
39128            2020-10-18      194bf3       2       4          11     2   
39129            2020-10-18      1b4b5a       2       4          26     2   
...                     ...         ...     ...     ...         ...   ...   
2183266          2020-10-18      2edfde       2       6          28     2   
2183267          2020-10-18      332cf5       2       4          30     1   
2183268          2020-10-18      42c51a       2      12           9     2   
2183270          2020-10-18      372368       2       9           8     2   
2183271          2020-10-18      427b3f       2       9          19     2   

         ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
39125             10           19             46              1  ...   
39126             14           14            120              1  ...   
39127             19           19             26              1  ...   
39128             11           11              7              1  ...   
39129             26           26             55              1  ...   
...              ...          ...            ...            ...  ...   
2183266           28           28              9              1  ...   
2183267           30           30            135              1  ...   
2183268            9            9             10              1  ...   
2183270            8            8             19              2  ...   
2183271           19           19             18              2  ...   

        RENAL_CRONICA TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  \
39125               2          2        99             2             97   
39126               2          2        99             2             97   
39127               2          2        99             2             97   
39128               2          2        99             2             97   
39129               2          2        99             2             97   
...               ...        ...       ...           ...            ...   
2183266             2          2         2             1              3   
2183267             2          2         2             1              3   
2183268             2          2         1             1              3   
2183270             2          2         2             1              3   
2183271             2          2         2             1              3   

         CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
39125                      6        99            México           97   97  
39126                      6        99            México           97   97  
39127                      6        99            México           97   97  
39128                      6        99            México           97   97  
39129                      6        99            México           97   97  
...                      ...       ...                ...          ...  ...  
2183266                    6        99            México           97   97  
2183267                    6        99            México           97   97  
2183268                    6        99            México           97   97  
2183270                    6        99            México           97    2  
2183271                    6        99            México           97    2  

[265492 rows x 38 columns]

In [26]:
sosp_sinm=data[(data["CLASIFICACION_FINAL"]==6)&(data["RESULTADO_LAB"]==97)]
sosp_sinm

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
39125            2020-10-18      193243       2       4          19     2   
39126            2020-10-18      0462c9       2       4          14     1   
39127            2020-10-18      0cafed       2       4          19     2   
39128            2020-10-18      194bf3       2       4          11     2   
39129            2020-10-18      1b4b5a       2       4          26     2   
...                     ...         ...     ...     ...         ...   ...   
2183248          2020-10-18      37c6a8       2      12          11     1   
2183251          2020-10-18      28b4d5       1      12           1     2   
2183255          2020-10-18      414a3f       1      12           1     2   
2183263          2020-10-18      3d7bff       1       4          19     2   
2183264          2020-10-18      2d9933       1      12          13     2   

         ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
39125             10           19             46              1  ...   
39126             14           14            120              1  ...   
39127             19           19             26              1  ...   
39128             11           11              7              1  ...   
39129             26           26             55              1  ...   
...              ...          ...            ...            ...  ...   
2183248           11           11             31              1  ...   
2183251            1            1              1              1  ...   
2183255            1            1              1              1  ...   
2183263           19           19             31              1  ...   
2183264           13           13             77              1  ...   

        RENAL_CRONICA TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  \
39125               2          2        99             2             97   
39126               2          2        99             2             97   
39127               2          2        99             2             97   
39128               2          2        99             2             97   
39129               2          2        99             2             97   
...               ...        ...       ...           ...            ...   
2183248             2          2         2             2             97   
2183251             2          2         2             2             97   
2183255             2          2        99             2             97   
2183263             2          2         2             2             97   
2183264             2          2         2             2             97   

         CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
39125                      6        99            México           97   97  
39126                      6        99            México           97   97  
39127                      6        99            México           97   97  
39128                      6        99            México           97   97  
39129                      6        99            México           97   97  
...                      ...       ...                ...          ...  ...  
2183248                    6        99            México           97   97  
2183251                    6        99            México           97   97  
2183255                    6        99            México           97   97  
2183263                    6        99            México           97   97  
2183264                    6        99            México           97   97  

[176840 rows x 38 columns]

In [33]:
sosp_sinpr=data[(data["CLASIFICACION_FINAL"]==6)&(data["RESULTADO_LAB"].isin([4))]
sosp_sinpr

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
39256            2020-10-18      2c4777       2       9           9     2   
39713            2020-10-18      0f2dd8       1      13          21     1   
40000            2020-10-18      10e9a3       1      12          10     2   
41098            2020-10-18      01bcdf       1      12          21     2   
41341            2020-10-18      13ec48       1      12          26     1   
...                     ...         ...     ...     ...         ...   ...   
2183266          2020-10-18      2edfde       2       6          28     2   
2183267          2020-10-18      332cf5       2       4          30     1   
2183268          2020-10-18      42c51a       2      12           9     2   
2183270          2020-10-18      372368       2       9           8     2   
2183271          2020-10-18      427b3f       2       9          19     2   

         ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
39256              9            9             14              1  ...   
39713             21           21             74              1  ...   
40000             10           10              5              1  ...   
41098             30           21            114              1  ...   
41341             26           26             30              1  ...   
...              ...          ...            ...            ...  ...   
2183266           28           28              9              1  ...   
2183267           30           30            135              1  ...   
2183268            9            9             10              1  ...   
2183270            8            8             19              2  ...   
2183271           19           19             18              2  ...   

        RENAL_CRONICA TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  \
39256               2          2         2             1              4   
39713               2          2         2             1              4   
40000               2          2        99             1              4   
41098               2          2         2             1              4   
41341               2          2         1             1              4   
...               ...        ...       ...           ...            ...   
2183266             2          2         2             1              3   
2183267             2          2         2             1              3   
2183268             2          2         1             1              3   
2183270             2          2         2             1              3   
2183271             2          2         2             1              3   

         CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
39256                      6        99            México           97   97  
39713                      6        99            México           97   97  
40000                      6        99            México           97   97  
41098                      6        99            México           97   97  
41341                      6        99            México           97   97  
...                      ...       ...                ...          ...  ...  
2183266                    6        99            México           97   97  
2183267                    6        99            México           97   97  
2183268                    6        99            México           97   97  
2183270                    6        99            México           97    2  
2183271                    6        99            México           97    2  

[88652 rows x 38 columns]

In [28]:
sosp_conpr=data[(data["CLASIFICACION_FINAL"]==6)&(data["RESULTADO_LAB"]==3)]
sosp_conpr

FECHA_ACTUALIZACION ID_REGISTRO  ORIGEN  SECTOR  ENTIDAD_UM  SEXO  \
2134604          2020-10-18      3611b6       2       4          15     2   
2134605          2020-10-18      2ba8f7       2       4          11     2   
2134606          2020-10-18      45a62c       2       4           9     1   
2134607          2020-10-18      23f242       2       4           9     1   
2134608          2020-10-18      28907e       2       4          15     1   
...                     ...         ...     ...     ...         ...   ...   
2183266          2020-10-18      2edfde       2       6          28     2   
2183267          2020-10-18      332cf5       2       4          30     1   
2183268          2020-10-18      42c51a       2      12           9     2   
2183270          2020-10-18      372368       2       9           8     2   
2183271          2020-10-18      427b3f       2       9          19     2   

         ENTIDAD_NAC  ENTIDAD_RES  MUNICIPIO_RES  TIPO_PACIENTE  ...  \
2134604           15           15             50              1  ...   
2134605           11           11              7              1  ...   
2134606           14           15            104              1  ...   
2134607            9            9             13              1  ...   
2134608           15           15            120              1  ...   
...              ...          ...            ...            ...  ...   
2183266           28           28              9              1  ...   
2183267           30           30            135              1  ...   
2183268            9            9             10              1  ...   
2183270            8            8             19              2  ...   
2183271           19           19             18              2  ...   

        RENAL_CRONICA TABAQUISMO OTRO_CASO  TOMA_MUESTRA  RESULTADO_LAB  \
2134604             2          2         2             1              3   
2134605             2          2         2             1              3   
2134606             2          2         2             1              3   
2134607             2          2         1             1              3   
2134608             2          2         1             1              3   
...               ...        ...       ...           ...            ...   
2183266             2          2         2             1              3   
2183267             2          2         2             1              3   
2183268             2          2         1             1              3   
2183270             2          2         2             1              3   
2183271             2          2         2             1              3   

         CLASIFICACION_FINAL  MIGRANTE  PAIS_NACIONALIDAD  PAIS_ORIGEN  UCI  
2134604                    6        99            México           97   97  
2134605                    6        99            México           97   97  
2134606                    6        99            México           97   97  
2134607                    6        99            México           97   97  
2134608                    6        99            México           97   97  
...                      ...       ...                ...          ...  ...  
2183266                    6        99            México           97   97  
2183267                    6        99            México           97   97  
2183268                    6        99            México           97   97  
2183270                    6        99            México           97    2  
2183271                    6        99            México           97    2  

[37200 rows x 38 columns]